corona-warn-app/cwa-app-android
# QuotaCalculator v2

## Description
The previous one had to be removed due to unforeseen side-effects in relation to transaction rollbacks.

* Created a new `ENFClient` which currently only has a `DiagnosisKeyProvider` module, but will in the future have further modules until the singleton `InternalExposureNotificationClient` is empty.
* Added additional logging statements to better understand transaction failures via logfile.

## How to test
* Review the code
* Think about whether we have covered all quota edge cases in `SubmissionQuotaTest`
* Run the app, force key retrieval via test menu, observe logcat (filter for `SubmissionQuota`)
